conans/paths/package_layouts/package_cache_layout.py

```diff
@@ -110,6 +110,43 @@ def package(self, pref):
         assert pref.ref == self._ref, "{!r} != {!r}".format(pref.ref, self._ref)
         return os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
 
+    @contextmanager
+    def set_dirty_context_manager(self, pref):
+        pkg_folder = os.path.join(self._base_folder, PACKAGES_FOLDER, pref.id)
+        set_dirty(pkg_folder)
```

Comment:
What if the package is already _dirty_? Should we raise an exception? Same comment for the `package_set_dirty` method. Not for this PR because it wouldn't be a refactor, but wdyt?

Reply:
yes, most likely, it would be great to add at least an assert, as in theory it shouldn't be possible. But an exception for readability from users would be good too, just in case.